encodings/dict/src/compute/binary_numeric.rs

```
@@ -0,0 +1,29 @@
+use vortex_array::array::ConstantArray;
+use vortex_array::compute::{binary_numeric, BinaryNumericFn};
+use vortex_array::{ArrayData, IntoArrayData};
+use vortex_error::VortexResult;
+use vortex_scalar::BinaryNumericOperator;
+
+use crate::{DictArray, DictEncoding};
+
+impl BinaryNumericFn<DictArray> for DictEncoding {
+    fn binary_numeric(
+        &self,
+        array: &DictArray,
+        rhs: &ArrayData,
+        op: BinaryNumericOperator,
+    ) -> VortexResult<Option<ArrayData>> {
```

We need a pre-check to verify that the values of the dictionary array are of a primitive type. If not, we can fast-fail to avoid creating a new ConstantArray.

---

Hmmm, maybe it's okay because we could assume that users should know that and this bad case is rare.

---

The so-called "entry-point" function should validate the DType of the arguments so each implementation doesn't have to do the same work

---

If I understand correctly, for every compute function, the `impl` for a particular encoding should only be called from the top-level function in `vortex-array/src/compute/`; therefore, we need not perform any checks that are done there. The `binary_numeric` from there checks the types and fails fast.